src/XMakeBuildEngine/Microsoft.Build.csproj

```diff
@@ -356,8 +356,8 @@
     <Compile Include="Definition\ResolvedImport.cs" />
     <Compile Include="Definition\SubToolset.cs" />
     <Compile Include="Definition\Toolset.cs" />
-    <Compile Include="Definition\ToolsetConfigurationReader.cs" />
-    <Compile Include="..\Shared\ToolsetElement.cs">
+    <Compile Include="Definition\ToolsetConfigurationReader.cs" Condition="$(FeatureSystemConfiguration) == 'true'"/>
+    <Compile Include="..\Shared\ToolsetElement.cs" Condition="$(FeatureSystemConfiguration) == 'true'">
```

Comment:
So far what I've been doing is that if the whole file needs to be included or excluded based on a feature, I would do that with a condition in MSBuild.  For groups of files like are in the SharedUtilities/Compat folder that feels better to me.  For one-off (or two-off) files like this one I don't have much of a strong opinion.

I'm going to go ahead and merge this, we can always discuss and make a change later.